### ISSUE TITLE
Single Option / Variant / Add to Cart Shopify Issue

### ISSUE THREAD

**Issue by m-w-d-d** [vercel/commerce#261]
Hi there,

Really appreciate your work on this! Have set up a Shopify link to a cloned version of the this repo - all seems to have connected ok but have come up with a couple of issues on the product page:

- If the product has no options/variants it displays 'Title' and you cannot add a product to cart as the button is disabled
- If the product has one variant option you cannot add a product to cart as the button is disabled
- The add to cart button only becomes enabled when a product has 2 variant options and both are selected.

As I say I have set this up almost out of the box and have only updated .env details and added `"provider": "shopify",` to the commerce.config.json file.

If I'm missing something obvious do let me know, I just thought I would ask before I start going in and moving code around!

Thanks again and I appreciate your time on this.

**Comment by Jeremy-Logan** [comment]
I had this problem too and noticed that there is a prop "disabled={!variant && product.options.length > 0}" in the cart button in components/product/ProductView/ProductView.tsx. After commenting the prop out the cart works fine. Couldn't tell you if taking it out has other side effects though.

**Comment by clockelliptic** [comment]
This could be some sort of data type error.

Check `ProductView.tsx` line 32, the state setter `setChoices`:

```js
  const [choices, setChoices] = useState<SelectedOptions>({
    size: null,
    color: null,
  })
```

You may need to update the initial state object to match the actual choices available for the given product. 

In which case you would also have to update the typescript helper definition for `SelecedOptions` found in `../helpers.ts`.

**Comment by okbel** [comment]
Can you pull the latest we fixed this in #222 ! Keep us posted!

**Comment by m-w-d-d** [comment]
@okbel - Looks like that fixed it.

Many thanks!
